snapcraft/internal/lxd.py

```diff
@@ -139,6 +144,61 @@ def _setup_project(self):
         self._push_file(tar_filename, dst)
         self._container_run(['tar', 'xvf', os.path.basename(tar_filename)])
 
+    def _inject_snapcraft(self):
+        if common.is_snap():
+            # Because of https://bugs.launchpad.net/snappy/+bug/1628289
+            self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
+
+            # Push core snap into container
+            self._inject_snap('core')
+            self._inject_snap('snapcraft', rev=os.environ.get('SNAP_REVISION'))
+        else:
+            self._container_run(['apt-get', 'install', 'snapcraft', '-y'])
+
+    def _inject_snap(self, name, *, rev=None):
+        session = requests_unixsocket.Session()
+        snapd_socket = '/run/snapd.socket'.replace('/', '%2F')
+        api = 'http+unix://{}/v2/snaps/{}'.format(snapd_socket, name)
+        try:
+            json = session.request('GET', api).json()
+        except Exception as e:
+            raise RuntimeError('Error connecting to {}'.format(api)) from e
+        if not json['status'] == 'OK':
+            raise RuntimeError('Error querying {} snap: {}'.format(name, json))
```

Comment:
These are pretty snapcraft-internal exceptions. Should we be using an exception inheriting from `SnapcraftError`?